builder/Dockerfile.fedora-gcc14-ubsan

```diff
@@ -0,0 +1,91 @@
+
+FROM fedora:40
```

Comment:
Do we need a new R build for this? Can't we reuse one of the existing builds that we already have? Also, if you need an R build, you'll need to update the docker compose file as well.